### PR TITLE
Docs: Update deployment docs

### PR DESCRIPTION
**Pull request by anna-geller**
### Summary
- Adding new CLI flags
- Adjusting to the new --work-queue behavior and tags

This pull request is:

- [x] A documentation / typographical error fix
	- No tests or issue needed

### CODE REVIEW

File: docs/concepts/deployments.md [hunk 1]
```diff
@@ -42,21 +40,24 @@ When creating a deployment, a user must answer *two* basic questions:
 A deployment additionally enables you to:
 
 - Schedule flow runs
-- Assign tags for filtering flow runs on work queues and in the Prefect UI
+- Assign `--work-queue` name to delegate flow runs to work queues
```
Comment (cicdw) [inline docs/concepts/deployments.md:43]: This list reads like CLI documentation, I think these should only say the concept names.  For example, "Assign a work queue name to delegate..." and "Assign one or multiple tags to organize..."

File: docs/concepts/deployments.md [hunk 2]
```diff
@@ -42,21 +40,24 @@ When creating a deployment, a user must answer *two* basic questions:
...
 - Assign custom parameter values for flow runs based on the deployment
 - Create ad-hoc flow runs from the API or Prefect UI
 - Upload flow files to a defined storage location for retrieval at run time
 
-Deployments can package your flow code; either Prefect Cloud or a local Prefect Orion server run with `prefect orion start`. 
+!!! warning "Flow run and work queue affinity improved starting from 2.0.5"
+    Until Prefect 2.0.4 `tags` were used to delegate flow runs created from deployments to the work queue agents that should pick up those runs. Given the flexibility of tags, it wasn't always clear which runs would be picked up by which work queue agent. To disambiguate this process, the tags now serve purely organizational purposes. To assign deployments to a relevant work-queue, use the `--work-queue` argument on your `prefect deployment build` command, e.g. `--work-queue k8s`. Then, to start an agent which can pick up such deployment, use the command: `prefect agent start k8s`.
```
Comment (cicdw) [inline docs/concepts/deployments.md:50]: Similarly here - I wouldn't assume a CLI user and think that _how_ to assign a work queue can be reserved for other parts of the doc outside of this warning

File: docs/concepts/deployments.md [hunk 3]
```diff
@@ -153,24 +154,31 @@ For example:
...
 |  -sb, --storage-block TEXT     | The storage block to use in `type/name` format. |
+|  -a, --apply     | Set this flag if you want to simultabeously build and apply the deployment within a single command. |
+|  --cron TEXT     | A cron string that will be used to set a CronSchedule on the deployment, e.g. `--cron "*/1 * * * *"` to create flow runs from that deployment every minute. |
+|  --interval INTEGER     | An integer specifying an interval (in seconds) that will be used to set an IntervalSchedule on the deployment, e.g. `--interval 60` to create flow runs from that deployment every minute. |
+|  --rrule TEXT     | An RRule that will be used to set an RRuleSchedule on the deployment, e.g. `--rrule 'FREQ=HOURLY;BYDAY=MO,TU,WE,TH,FR;BYHOUR=9,10,11,12,13,14,15,16,17'` to create flow runs from that deployment every hour but only during business hours. |
 
 When you run this command, Prefect: 
 
-- Creates the and `catfacts_flow-deployment.yaml` file for your deployment based on your flow code and options.
+- Creates the and `marvin_flow-deployment.yaml` file for your deployment based on your flow code and options.
```
Comment (cicdw) [inline docs/concepts/deployments.md:181]:
```suggestion
- Creates a `marvin_flow-deployment.yaml` file for your deployment based on your flow code and options.
```

**Review by cicdw — COMMENTED** [review]
a few suggestions

File: docs/concepts/deployments.md [hunk 4]
```diff
@@ -19,49 +18,78 @@ tags:
...
+- Schedule flow runs.
+- Assign a work queue name to delegate deployment flow runs to work queues.
+- Assign one or more tags to organize your deployments and flow runs. You can use those tags as filters in the [Prefect UI](/ui/flow-runs/).
+- Assign custom parameter values for flow runs based on the deployment.
+- Create ad-hoc flow runs from the API or Prefect UI.
+- Upload flow files to a defined storage location for retrieval at run time.
+- Specify run time infrastructure for flow runs, such as Docker or Kubernetes configuration.
 
-Deployments can package your flow code; either Prefect Cloud or a local Prefect Orion server run with `prefect orion start`. 
+With storage blocks, you can package not only your flow code script but also any supporting files, including your custom modules, SQL scripts and any configuration files needed in your project.
```
Comment (cicdw) [inline docs/concepts/deployments.md:78]:
```suggestion
With remote storage blocks, you can package not only your flow code script but also any supporting files, including your custom modules, SQL scripts and any configuration files needed in your project.
```

In an effort to send the right signal with our names, I've been trying to replace conversational references to "storage" with "remote storage" to emphasize that they are only useful in some sort of distributed / remote context.

File: docs/concepts/deployments.md [hunk 5]
```diff
@@ -19,49 +18,78 @@ tags:
...
 
+The following diagram provides a high-level overview of the conceptual elements involved in defining a deployment and executing a flow run based on that deployment.
+
+```mermaid
+graph LR
+    F(Flow Code):::yellow -.-> A(Deployment Definition):::gold
+    subgraph Server [Prefect API]
+    D(Deployment):::green
+    end
+    subgraph Storage
```
Comment (cicdw) [inline docs/concepts/deployments.md:42]:
```suggestion
    subgraph Remote Storage
```
(see comment below)

File: docs/concepts/deployments.md [hunk 6]
```diff
@@ -79,11 +107,26 @@ graph LR
 
 This enables you to run a single flow with different parameters, on multiple schedules, and in different environments. This also enables you to run different versions of the same flow for testing and production purposes.
 
-## deployment.yaml
+## Deployment definition
+
+A _deployment definition_ captures the settings for creating a [deployment object](#deployment-api-representation) on the Prefect API. You can create the deployment definition by:
+
+- Run the [`prefect deployment build` CLI command](#create-a-deployment) CLI command with deployment options to create a [`deployment.yaml`](#deploymentyaml) deployment definition file, then run `prefect deployment apply` to create a deployment on the API using the settings in `deployment.yaml`.
```
Comment (cicdw) [inline docs/concepts/deployments.md:114]:
```suggestion
- Run the [`prefect deployment build` CLI command](#create-a-deployment) with deployment options to create a [`deployment.yaml`](#deploymentyaml) deployment definition file, then run `prefect deployment apply` to create a deployment on the API using the settings in `deployment.yaml`. 
```

File: docs/concepts/deployments.md [hunk 7]
```diff
@@ -19,49 +18,78 @@ tags:
...
-Each deployment references a single flow (though that flow may, in turn, call any number of tasks and subflows). Any single flow, however, may be referenced by any number of deployments. 
+Each deployment references a single "entrypoint" flow (though that flow may, in turn, call any number of tasks and subflows). Any single flow, however, may be referenced by any number of deployments. 
 
 At a high level, you can think of a deployment as configuration for managing flows, whether you run them via the CLI, the UI, or the API.
 
 !!! warning "Deployments have changed since beta"
-    Deployments now rely on deployment YAML files for specifying a deployment of your workflow.
-
-    Deployments based on `Deployment` and `DeploymentSpec` are no longer supported.
+    Deployments based on `DeploymentSpec` are no longer supported. Instead, you can define deployments by using either the [`prefect deployment` CLI command](#create-a-deployment) or the `Deployment` Python object.
```
Comment (cicdw) [inline docs/concepts/deployments.md:26]:
```suggestion
    Deployments based on `DeploymentSpec` are no longer supported. Instead, you can define deployments by using either the [`prefect deployment` CLI command](#create-a-deployment) or [the `Deployment` Python object](/api-ref/prefect/deployments/).
```

File: docs/concepts/deployments.md [hunk 8]
```diff
@@ -134,50 +177,61 @@ parameter_openapi_schema:
 
 To create a deployment from an existing flow script, there are two steps:
 
-1. Build the `deployment.yaml` file, which includes uploading your flow to its configured storage location.
+1. Build the deployment definition file `deployment.yaml`. This step includes uploading your flow to its configured storage location, if one is specified.
```
Comment (cicdw) [inline docs/concepts/deployments.md:180]:
```suggestion
1. Build the deployment definition file `deployment.yaml`. This step includes uploading your flow to its configured remote storage location, if one is specified.
```

File: docs/concepts/deployments.md [hunk 9]
```diff
@@ -306,7 +360,7 @@ $ prefect deployment inspect 'Cat Facts/catfact'
 
 ## Running deployments
 
-If you specify a schedule for a deployment, the deployment will execute its flow automatically on that schedule as long as a Prefect Orion API server and agent is running.
+If you specify a schedule for a deployment, the deployment will execute its flow automatically on that schedule as long as a Prefect Orion API server and agent is running. Prefect Cloud can create scheduled flow runs automatically as long as an agent is configured to pick up flow runs for the deployment.
```
Comment (cicdw) [inline docs/concepts/deployments.md:363]: The new sentence is not quite right -- the scheduler will run no matter what and create new runs, but they won't actually run unless an agent is running that can pick them up.

**Review by cicdw — COMMENTED** [review]
a few super minor suggestions, otherwise LGTM!

**Comment by anna-geller** [comment]
I can't approve a PR I initiated, but @tpdorsey I approve all the changes you made, looks great now - added a couple of nits via Slack

**Review by cicdw — APPROVED** [review]

**Pull request merged**